### ISSUE TITLE
[Chore]: Add extensive test cases (e2e)

### ISSUE THREAD

**Issue by tiffanosaurus** [netlify/netlify-plugin-nextjs#839]
Parent issue to keep track of all test cases/sites we want to have covered via Cypress. We will want to test most features of Next to ensure we don't break anything. Feel free to add more test cases.

## Default site
- [X] Set up cypress
- [x] Default site
- [x] `basePath`
- [x] Preview mode
- [x] i18n
- [x] Headers
- [x] Rewrites/redirects
- [x] `basePath` + i18n
- [x] Trailing slash
- [x] Catch-all/optional catch-all, getxxprops
- [x] Images
- [x] Dynamic imports
- [x] Custom 404/500
- [x] Static assets in `public`
- [x] cookie header locale
- [ ] middleware
- [ ] advanced rewrite rules

## Monorepo site
- [ ] Custom dist
- [ ] Monorepo: `next` run in root with site root in subdirectory

## Nx site
- [ ] nx create-app

**Comment by nickytonline** [comment]
@tiffanosaurus, I'm going to go ahead and remove you from this as you've switched teams. Whoever picks this up, there's still some E2E tests to complete.

**Comment by nickytonline** [comment]
Hi @netlify/frameworks! We've been working on the Next.js plugin quite a bit lately. Just wondering if any of that work involved E2E tests for the middleware and advanced rewrite rules?

**Comment by orinokai** [comment]
Hey @nickytonline, no afaik we haven't added tests yet for those features. I'm hoping to chat to Erica later this week about a few testing things so I'll roll this into it.

**Comment by MarcL** [comment]
@orinokai - Where are we at with this work? I was wondering if it's worth splitting this up into more manageable tickets if there's lots left to do.

**Comment by orinokai** [comment]
I had planned to give the tests a good review after I finished the fallback work because the tests would need to be updated, but both aspects of that work are currently blocked and they should probably be a separate issue once the work is complete away. I'll make a start on this ticket this week.
